src/protocols/bdx/TransferFacilitator.cpp

```diff
@@ -78,7 +78,10 @@ void TransferFacilitator::PollForOutput()
     HandleTransferSessionOutput(outEvent);
 
     VerifyOrReturn(mSystemLayer != nullptr, ChipLogError(BDX, "%s mSystemLayer is null", __FUNCTION__));
-    mSystemLayer->StartTimer(mPollFreq, PollTimerHandler, this);
+    if (!mStopPolling)
+        mSystemLayer->StartTimer(mPollFreq, PollTimerHandler, this);
+    else
+        mSystemLayer->CancelTimer(PollTimerHandler, this);
```

Comment:
Suggest the following, since it's otherwise not clear that a transfer facilitator cannot be reused.

```suggestion
    if (!mStopPolling)
    {
        mSystemLayer->StartTimer(mPollFreq, PollTimerHandler, this);
    }
    else
    {
        mSystemLayer->CancelTimer(PollTimerHandler, this);
        mStopPolling = false;
    }
```